static/js/common/EditableText.js

```diff
@@ -0,0 +1,182 @@
+};
+
+const centerDiv = {
+    margin: 'auto',
+    width: '10%',
+};
+
+class EditableText extends Component {
+
+    static defaultProps = {
+        fieldName: '',
+        status: '',
+        onSave: null,
+        style: {},
+        text: '',
+        type: 'input',
+    }
+    static propTypes = {
+        status: PropTypes.string,
+        fieldName: PropTypes.string,
+        onSave: PropTypes.func,
+        style: PropTypes.object,
+        text: PropTypes.string,
+        type: PropTypes.oneOf(['textarea', 'input']),
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            editing: false,
+            saving: false,
+            text: this.props.text,
+            message: '',
+        };
+
+        this.edit = this.edit.bind(this);
+        this.saveEdit = this.saveEdit.bind(this);
+        this.renderSavingAnimation = this.renderSavingAnimation.bind(this);
+        this.cancelEdit = this.cancelEdit.bind(this);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.status === 'success') {
+            this.setState({
+                message: 'Saved',
+                saving: false,
+                editing: false });
+            wait(3000)
+            .then(() => {
+                this.setState({ message: '' });
+            });
+        } else if (nextProps.status === 'error') {
+            this.setState({
+                message: "Couldn't save, try again later",
+                saving: false,
+                editing: false });
+            wait(3000)
+            .then(() => {
+                this.setState({ message: '' });
+            });
+        }
+    }
+
+    cancelEdit() {
+        this.setState({ editing: false, message: '' });
+    }
+
+    edit() {
+        this.setState({ editing: true });
+    }
+
+    saveEdit() {
+        this.setState({
+            saving: true,
+        });
+        if (this.props.type === 'input' && this.input.value.length > 20) {
+            this.setState({
+                message: 'Field cannot be longer than 20 symbols',
+                saving: false,
+                editing: false });
+            wait(3000)
+            .then(() => {
+                this.setState({ message: '' });
+            });
+        } else if (this.props.type === 'textarea' && this.input.value.length > 200) {
+            this.setState({
+                message: 'Field cannot be longer than 200 symbols',
+                saving: false,
+                editing: false });
+            wait(3000)
+            .then(() => {
+                this.setState({ message: '' });
+            });
+        } else {
+            this.props.onSave(this.props.updateId, this.props.fieldName, this.input.value);
+            this.setState({text: this.input.value});
+        }
+    }
+
+    renderSavingAnimation() {
+        if (this.state.saving) {
+            return <ReactLoading style={centerDiv} type="bars" color="#444" height="70" width="20" />;
```

Comment:
move repeatable values into constants.
I've seen `bars` and color `#444` many times

Reply:
This is temporary for now. In the future, we will create new preloader